File: squares-of-a-sorted-array/squares-of-a-sorted-array.cs
public class Solution {
    public int[] SortedSquares(int[] nums) {
        
        int[] result=new int[nums.Length];
        
        int first=0;
        int last=nums.Length-1;
        int ind=last;
        
        while(first<=last && ind>=0)
        {
            if(Math.Abs(nums[first])<Math.Abs(nums[last]))
            {
               result[ind]=(int)Math.Pow(nums[last],2);
                last--;
            }
            else
            {
                result[ind]=(int)Math.Pow(nums[first],2);
                first++;
            }
            ind--;
        }
        
        return result;
    }
}

File: squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        
        
        int[] result=new int[nums.length];
        
        int left=0;
        int right=nums.length-1;
        
        int k=nums.length-1;
        while(left<=right && k>=0)
        {
            if(Math.abs(nums[left]) > Math.abs(nums[right]))
            {
                result[k]=nums[left]*nums[left];
                left++;
            }
            else
            {
                result[k]=nums[right]*nums[right];
                right--;
            }
            k--;
            
        }
        
        return result;
        
    }
}